CMakeLists.txt

```diff
@@ -1,47 +1,92 @@
-# Build protocol buffers just to check they compile without errors.
-
 cmake_minimum_required(VERSION 2.8)
-add_definitions(-std=c++11)
-project(build_protos C CXX)
 
-find_package(Protobuf 3 REQUIRED)
+project(matrixio_protos C CXX)
 
-# For GRPC.
+# Load GRPC macro
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
+
+add_definitions(-std=c++11)
+
+# Required packages to build
+find_package(Protobuf 3 REQUIRED)
 find_package(GRPC REQUIRED)
 
-# vision.proto
-PROTOBUF_GENERATE_CPP(VISION_PROTO_SRC VISION_PROTO_HEADER vision/vision.proto)
+# protoc program
+find_program(PROTOC NAMES protoc)
+if (NOT EXISTS ${PROTOC})
+  message(FATAL_ERROR "The protoc program was not found")
+endif()
+message(STATUS "Found protoc program: " ${PROTOC})
 
-# GRPC vision_service
-PROTOBUF_GENERATE_CPP(VISION_GRPC_PROTO_SRC VISION_GRPC_PROTO_HEADER vision/vision_service.proto)
-PROTOBUF_GENERATE_GRPC_CPP(VISION_GRPC_SRC VISION_GRPC_HEADER vision/vision_service.proto)
+# Language to generate
+if (NOT DEFINED GEN_LANG)
+  set(GEN_LANG "cpp")
+endif()
 
-# GRPC recognition_service
-PROTOBUF_GENERATE_CPP(RECOGNITION_GRPC_PROTO_SRC RECOGNITION_GRPC_PROTO_HEADER vision/recognition_service.proto)
-PROTOBUF_GENERATE_GRPC_CPP(RECOGNITION_GRPC_SRC RECOGNITION_GRPC_HEADER vision/recognition_service.proto)
-PROTOBUF_GENERATE_CPP(RECOGNITION_PROTO_SRC RECOGNITION_PROTO_HEADER vision/recognition.proto)
+# protoc grpc plugin
+find_program(GRPC_PLUGIN NAMES "grpc_${GEN_LANG}_plugin")
+if (NOT EXISTS ${GRPC_PLUGIN})
+  message(WARNING "The grpc_${GEN_LANG}_plugin plugin was not found, \
```

Comment:
We could, but then there are low level developers that only use protobuf, and may not want to compile all the gRPC stuff.